src/main/java/com/immunopass/repository/VoucherRepository.java

```diff
@@ -1,11 +1,32 @@
 package com.immunopass.repository;
 
 import com.immunopass.entity.VoucherEntity;
+import com.immunopass.enums.VoucherStatus;
+import com.immunopass.model.Voucher;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 
+import java.util.List;
 import java.util.Optional;
 
 
 public interface VoucherRepository extends JpaRepository<VoucherEntity, Long> {
+    @Modifying
+    @Query("update voucher set status = :status where order_id = :orderID")
+    void updateVoucherStatusForOrder(VoucherStatus status, Long orderID);
+
+    @Query("SELECT * FROM voucher WHERE order_id = :orderID")
```

Comment:
You can avoid @query annotation here if you follow proper method naming convention.